File: 2114_maximum_number_of_words_found_in_sentences/solution.go
// LeetCode 2114 - Maximum Number of Words Found in Sentences
// https://leetcode.com/problems/maximum-number-of-words-found-in-sentences/

import "strings"

func mostWordsFound(sentences []string) int {
	ans := 0
	for _, s := range sentences {
		c := strings.Count(s, " ") + 1
		if c > ans {
			ans = c
		}
	}
	return ans
}

File: 2114_maximum_number_of_words_found_in_sentences/solution.c
// LeetCode 2114 - Maximum Number of Words Found in Sentences
// https://leetcode.com/problems/maximum-number-of-words-found-in-sentences/

int mostWordsFound(char** sentences, int sentencesSize) {
    int ans = 0;
    for (int i = 0; i < sentencesSize; i++) {
        int c = 1;
        for (char* p = sentences[i]; *p; p++) if (*p == ' ') c++;
        if (c > ans) ans = c;
    }
    return ans;
}
